libsel4vm/src/arch/arm/vgic/vgic.c

```diff
@@ -1115,28 +1115,17 @@ int vm_install_vgic(vm_t *vm)
     int err;
 
     vgic = calloc(1, sizeof(*vgic));
-    if (!vgic) {
-        assert(!"Unable to calloc memory for VGIC");
-        return -1;
-    }
+    assert(vgic);
```

Comment:
Not sure is you can do that, there is no global guarantee `calloc()` will be successful and asserts are removed in release code. Now one could argue that it never fails in release code because this is well tested - but still, you don't know the context this is used in. So I'd prefer to keep this and have an error path here that the caller must deal with.
Another option would be making all this static, so the compiler/linker can ensure this. How much dynamism do we need there in the end?

Reply:
I agree with @axel-h, the error checking should stay as this is library and needs to allow applications to make policy choices about whether to abort on errors or do something else.

Reply:
Yes, this was my mistake, I wrongly assumed that asserts were enabled in release mode. I'll make these changes once #28  goes through.